### DISCUSSION TITLE
Specifying a python env to use in a .qmd file

### DISCUSSION THREAD

**Discussion by Dpananos** [quarto-dev/quarto-cli#1214] [Q&A]
I'm using quarto to write my blog posts.  I would like to create a conda env for my latest posts because it uses some other frameworks and I don't want to bork my base env.

Is there a way to specify which conda env I want to use in the front matter if the index.qmd?  If not in the front matter, is there a way to do it at all?

**Comment by jjallaire** [comment]
Quarto doesn't activate conda environments itself, rather it runs within whatever conda environment it finds itself within. So I'd suggest just making sure that your shell or IDE is using the requisite environment before rendering.

**Reply by cscheid** [reply]
I think the issue here might be that some of their older posts will want older environments.

If that's the case case, I think the solution is to use `freeze: true`, and to render each file with its appropriate environment.

**Reply by mcanouil** [reply]
I wonder if it works for website or book to set `renv` (for R and Python) per quarto file? (I did not try yet 😇)

**Comment by chuymtz** [comment]
I have a work around that works for me but I'm not entirely sure this is the proper usage. I'm using vscode and what I always make sure of is that the kernel used in my current vscode session is the one I want to be used to render my qmd file. 

In vscode this can be done on the bottom right corner while having a .py file open and selecting the desised kernel. Similar to running ipykernel interactively.

**Comment by bluelephant825** [comment]
From the [documentation](https://quarto.org/docs/computations/python.html):

<img width="765" alt="Screenshot 2022-10-05 at 19 39 01" src="https://user-images.githubusercontent.com/4818315/194125901-304b97b5-9f94-4215-bf0f-cd125a71b56e.png">

In my case, I had installed python3 and jupyter on my Mac using Anaconda. So, the path to my python install was:

/Users/odebroqueville/opt/anaconda3/bin/python

but VS Code was using the python path provided by Xcode and therefore couldn't find jupyter.

Setting the correct path in the environment variable:

QUARTO_PYTHON=/Users/username/opt/anaconda3/bin/python

and restarting the terminal in VS Code fixed the problem. I used fig (from fig.io) to set the env var, but you could just as well set the env var in ~/.zshrc I guess.

<img width="1142" alt="Screenshot 2022-10-05 at 19 50 28" src="https://user-images.githubusercontent.com/4818315/194127795-490c4ff2-4f26-4bea-ac45-91b7760fc54c.png">

Here's the result in VS Code:

<img width="1392" alt="Screenshot 2022-10-05 at 19 52 43" src="https://user-images.githubusercontent.com/4818315/194128177-e7fdbe0c-be04-49c7-9900-a21ad09a6e7a.png">

**Reply by mcanouil** [reply]
Within python VSCode extensions, there are also settings to set Python path.

**Reply by bluelephant825** [reply]
You're absolutely right! Setting the Python Default Interpreter Path in Microsoft's Python extension would have the same effect as setting the above mentioned environment variable.

<img width="1392" alt="Screenshot 2022-10-05 at 21 11 55" src="https://user-images.githubusercontent.com/4818315/194145449-dcc3c9d2-578f-4e6b-8d5a-454a86d7def6.png">

**Comment by r-leyshon** [comment]
From the [quarto docs](https://quarto.org/docs/computations/python.html#kernel-selection), update the YAML with your local conda file created with the prefix flag.

```
jupyter: 
  kernelspec:
    name: "conda-env-<YOUR_ENV_NAME>-py"
    language: "python"
    display_name: "<YOUR_ENV_NAME>"
```

I put a [quick blog](https://thedatasavvycorner.com/blogs/08-quarto-conda-env) together to record it. Hope it helps someone.

**Reply by Ben-physics-dotcom** [reply]
Thank you a lot. I am learning Quarto in the moment and your post helped me a lot. I got error messages, if I render a notebook and with the kernelspec the error messages are solved.
